chingu-coders/Voyage2-Turtles-02
# fix bug in pomodoro clock

## Purpose
* Address bug in pomodoro clock, which causes clock to stop working when devTab is out of focus

## Approach/Fixes
* replace `requestAnimationFrame()` with `setTimeout()`, because `setTimout()` will run in background. This is not the most ideal solution, because it makes the UI a little laggy (you'll see that clock doesn't instantly respond to button clicks), but it is the solution that works. I thought about implementing a Web Worker, but WWs don't have access to `requestAnimationFrame()`, so we would have been using `setTimeout()` anyway.

## Testing
* Checkout this PR
* open a new tab and start a pomodoro cycle
* switch to new tab so that devTab loses focus
* confirm that alarm sounds at end of pomodoro cycle
* this works in chrome and firefox

NOTE: this PR is made to `development` branch. I think only @ziggysauce has ability to merge `development` with `master`
